### PR TITLE
Update/readme

### PR DESCRIPTION
**Pull request by jedwards4b**
### Description of changes
Update the readme file replacing manage-externals with git-fleximod

### Specific notes

Contributors other than yourself, if any:

Fixes: [Github issue #s] And brief description of each issue.

User interface changes?: [ No/Yes ]
[ If yes, describe what changed, and steps taken to ensure backward compatibility ]

Testing performed (automated tests and/or manual tests):

### CODE REVIEW

**Review by mnlevy1981 — CHANGES_REQUESTED** [review]
On line 97, we use `release-cesm2.1.2` as our example tag to checkout... can we update this to `cesm3_0_beta02` or `cesm3_0_alpha03b`? Maybe we could also mention that this process changed since CESM2 and include a link to the [CESM 2.1.5 README](https://github.com/ESCOMP/CESM/blob/release-cesm2.1.5/README.rst)?

**Comment by mnlevy1981** [comment]
(The "Switching to a different CESM tag" section around line 155 also talks about moving from 2.1.2 to 2.1.3 instead of switching between CESM3 tags)

**Comment by jedwards4b** [comment]
Fixed

File: README.rst [hunk 1]
```diff
@@ -219,27 +205,27 @@ If you'd like to modify a component via a branch and point to that
...
+#. Update **.gitmodules** to point to a hash on your branch. You can replace the
+   **tag** entry with a **hash** entry, as follows, note that we have also changed the url to
+   point to a personal fork::
+
+  [submodule "cam"]
+  path = components/cam
+  url = https://www.github.com/mycamfork/CAM
+  fxDONOTUSEurl = https://www.github.com/ESCOMP/CAM
+  fxtag = 94eaf83
+  fxrequired = ToplevelRequired
```
Comment (mnlevy1981) [inline README.rst:220]: This needs to be indented so it is treated as a code block, right now it's a mangled quote:

modifying component

Everything else looks great, thanks for updating this!

**Review by mnlevy1981 — APPROVED** [review]
Once https://github.com/ESCOMP/CESM/pull/283#discussion_r1744020027 has been addressed, this is ready to go

File: README.rst [hunk 2]
```diff
@@ -219,27 +205,27 @@ If you'd like to modify a component via a branch and point to that
...
+#. Create a CAM branch. Let's assume you have created this branch and called it
    **my_branch**.
 
-#. Update **Externals.cfg** to point to your branch. You can replace the
-   **tag** entry with a **branch** entry, as follows::
+#. Update **.gitmodules** to point to a hash on your branch. You can replace the
+   **tag** entry with a **hash** entry, as follows, note that we have also changed the url to
+   point to a personal fork::
+
+  [submodule "cam"]
```
Comment (mnlevy1981) [inline README.rst:215]: I think this line needs two more spaces (comparing it to lines [230-235 of the older version](https://github.com/ESCOMP/CESM/blob/cesm3_0_alpha03b/README.rst?plain=1#L230-L235))

**Review by mnlevy1981 — APPROVED** [review]
LGTM!

**Pull request merged**